mlir/include/mlir/Analysis/DataFlowFramework.h

```diff
@@ -429,10 +476,15 @@ class DataFlowSolver {
 
   /// A type-erased map of lattice anchors to associated analysis states for
   /// first-class lattice anchors.
-  DenseMap<LatticeAnchor, DenseMap<TypeID, std::unique_ptr<AnalysisState>>,
-           DenseMapInfo<LatticeAnchor::ParentTy>>
+  DenseMap<LatticeAnchor, DenseMap<TypeID, std::unique_ptr<AnalysisState>>>
       analysisStates;
 
+  /// A type-erased map of lattice type to the equivalet lattice anchors.
+  /// Lattice anchors are considered equivalent under a certain lattice type if
+  /// and only if, under this lattice type, the lattices pointed to by these
+  /// lattice anchors necessarily contain identical value.
+  DenseMap<TypeID, llvm::EquivalenceClasses<LatticeAnchor>> equivalentAnchorMap;
```

Comment:
It seems not a type-erased map. `type-erased` seems to refer to the `AnalysisState` as its child struct type info is erased.

TypeID used here is AnalysisState type. Calling it lattice type is kind of a step-forward I think as only in dense/sparse analysis we define dense/sparse lattice

Also typo on `equivalet`

```suggestion
  /// A map of analysis state type to the equivalence classes of lattice anchors.
  /// Lattice anchors are considered equivalent under a certain analysis state type if
  /// and only if the analysis states pointed to by these
  /// lattice anchors necessarily contain identical value.
  DenseMap<TypeID, llvm::EquivalenceClasses<LatticeAnchor>> equivalentAnchorMap;
```